### PR TITLE
Move all polls processing to events-sdk & prep for stable polls

### PR DESCRIPTION
**Pull request by turt2live**
**Requires https://github.com/matrix-org/matrix-js-sdk/pull/2102**
**Builds on https://github.com/matrix-org/matrix-react-sdk/pull/7462**

After this change, polls will support the full range of extensible events for both parsing and generation. The events-sdk has been specifically built to keep producing `m.text` variations of events though, for support with mobile's polls implementation.

This additionally fixes some bugs with the implementation - some cases of invalid event were being counted as unrenderable when in fact they are spoiled or similar.

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

Preview: https://61e1ea929f4ef329164f83c5--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/dialogs/EndPollDialog.tsx [hunk 1]
```diff
@@ -57,17 +56,10 @@ export default class EndPollDialog extends React.Component<IProps> {
...
         if (endPoll) {
-            const endContent: IPollEndContent = {
-                [POLL_END_EVENT_TYPE.name]: {},
-                "m.relates_to": {
-                    "event_id": this.props.event.getId(),
-                    "rel_type": "m.reference",
-                },
-                [TEXT_NODE_TYPE.name]: message,
-            };
+            const closure = PollEndEvent.from(this.props.event.getId(), message).serialize();
```
Comment (andybalaam) [inline src/components/views/dialogs/EndPollDialog.tsx:59]: I don't understand the variable name `closure`.  Would suggest `endEvent` maybe?

File: test/components/views/messages/MPollBody-test.tsx [hunk 2]
```diff
@@ -442,14 +460,14 @@ describe("MPollBody", () => {
...
-            return { "id": `id${i}`, "org.matrix.msc1767.text": `Name ${i}` };
+    it("renders the first 20 answers if 21 were given", () => {
+        const answers = Array.from(Array(21).keys()).map((i) => {
+            return { "id": `id${i}`, [M_TEXT.name]: `Name ${i}` };
         });
         const votes = [];
         const ends = [];
         const body = newMPollBody(votes, ends, answers);
-        expect(body.html()).toBe("");
+        expect(body).toMatchSnapshot();
```
Comment (andybalaam) [inline test/components/views/messages/MPollBody-test.tsx:470]: This makes the test much harder to read and understand what is happening.  Is there some more explicit assertion we can make?

File: test/components/views/messages/MPollBody-test.tsx [hunk 3]
```diff
@@ -442,14 +460,14 @@ describe("MPollBody", () => {
         expect(body.html()).toBe("");
     });
 
-    it("renders nothing if poll has more than 20 answers", () => {
-        const answers = [...Array(21).keys()].map((i) => {
-            return { "id": `id${i}`, "org.matrix.msc1767.text": `Name ${i}` };
+    it("renders the first 20 answers if 21 were given", () => {
```
Comment (andybalaam) [inline test/components/views/messages/MPollBody-test.tsx:463]: From the spec: `answers must be an array with at least 1 option and no more than 20. Lengths outside this range are invalid and must not be rendered by clients.`

I understood this to mean that the poll is invalid in this case?

**Review by andybalaam — CHANGES_REQUESTED** [review]
Looks great - one minor comment and one question to resolve.

File: test/components/views/messages/MPollBody-test.tsx (hunk 3 above)
Comment (turt2live) [reply test/components/views/messages/MPollBody-test.tsx:463]: The MSC was intending to say truncated, but clearly I didn't think to use that word 🤦‍♂️ 

"outside this range are invalid" means to say that values *within* the range are valid but anything outside of it is to be ignored.

File: test/components/views/messages/MPollBody-test.tsx (hunk 2 above)
Comment (turt2live) [reply test/components/views/messages/MPollBody-test.tsx:470]: possibly just a count of DOM elements, if I can figure out how to do that. Snapshot was just lazy and easy :p

File: test/components/views/messages/MPollBody-test.tsx (hunk 3 above)
Comment (turt2live) [reply test/components/views/messages/MPollBody-test.tsx:463]: I'll update the MSC to clarify.

**Review by andybalaam — APPROVED** [review]

**Pull request merged**